README.md

```diff
@@ -44,6 +57,153 @@ In most cases this command must be run by root user or through `sudo` command.
 
 BSD systems are not tested well. If ALSA works, Oto should work.
 
+## Usage
+
+The two main components of Oto are a `Context` and `Players`. The context handles interactions with
+the OS and audio drivers, and as such there can only be **one** context in your program.
+
+From a context you can create any number of different players, where each player is given an `io.Reader` that
+it reads bytes representing sounds from and plays.
+
+Note that a single `io.Reader` must **not** be used by multiple players.
+
+### Playing sounds from memory
+
+The following is an example of loading and playing an MP3 file:
+
+```go
+package main
+
+import (
+    "github.com/hajimehoshi/go-mp3"
+    "github.com/hajimehoshi/oto/v2"
+    "time"
+    "os"
+)
+
+func main() {
+
+    // Read the mp3 file into memory
+    fileBytes, err := os.ReadFile("./my-file.mp3")
+    if err != nil {
+        panic("Failed to load mp3")
+    }
+
+    //Convert the pure bytes into a reader object that can be used with the mp3 decoder
+    fileBytesReader := bytes.NewReader(fileBytes)
+
+    // Decode file
+    decodedMp3, err := mp3.NewDecoder(fileBytesReader)
+    if err != nil{
+        panic("Failed to decode mp3")
+    }
+
+    // Prepare an Oto context (this will use your default audio device) that will
+    // play all our sounds. Its configuration can't be changed later.
+
+    // Usually 44100 or 48000. Other values might cause distortions in Oto
+    samplingRate := 44100
+
+    // Number of channels (aka locations) to play sounds from. Either 1 or 2.
+    // 1 is mono sound, and 2 is stereo (most speakers are stereo). 
+    numOfChannels := 2
+
+    // Bytes used by a channel to represent one sample. Either 1 or 2 (usually 2).
+    audioBitDepth := 2
+
+    // Remember that you should **not** create more than one context
+    otoCtx, readyChan, err := oto.NewContext(samplingRate, numOfChannels, audioBitDepth)
+    if err != nil {
+        panic("Failed to create oto context")
+    }
+    // It might take a bit for the hardware audio devices to be ready, so we wait on the channel.
+    <-readyChan
+
+    // Create a new 'player' that will handle our sound. Paused by default.
+    player := otoCtx.NewPlayer(decodedMp3)
+    
+    // Play starts playing the sound and returns without waiting for it (Play() is async).
+    player.Play()
+
+    // We can wait for the sound to finish playing using something like this
+    for player.IsPlaying() {
+        time.Sleep(time.Millisecond)
+    }
+
+    // Now that the sound finished playing, we can restart from the beginning with these two lines
+    player.Reset()
+    player.Play()
```

Comment:
So, in this example, this play immediately stops, right? I think this seems confusing. What about removing this Reset and Play?

Reply:
Well I wanted to show how to rewind/replay. Maybe we can put the loop here again or a simple time.Sleep?

Reply:
Or just commenting out these line would be fine.

Reply:
Commented out 👍🏻